hash_practice/exercises.py

```diff
@@ -2,18 +2,36 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    pass
+    anagram_dict = {}
+
+    for string in strings: 
+        word_sorted = "".join(sorted(string))
+        if word_sorted not in anagram_dict: 
+            anagram_dict[word_sorted] = [string]
+        else: 
+            anagram_dict[word_sorted] = [string]

```

Comment:
Oof, this causes a pretty bad bug that means only the last string of each anagram group is returned, rather than all the strings in that group. I think you meant to `append` to the list here rather than replacing the value.

However, the solution is otherwise correct, so I'm only going to mark this Yellow. I will upgrade to Green upon this bug being fixed.